### ISSUE TITLE
OS interface changes for Solo5 break mirage-os-shim

### ISSUE THREAD

**Issue by mato** [mirage/mirage-os-shim#5]
Building mirage-skeleton static_website_tls against the latest Solo5 development package universe (OPAM remote at https://github.com/Solo5/opam-solo5) results in the following error:

````
+ ocamlfind ocamlopt -g -dontlink unix -dontlink str -dontlink num -dontlink threads -linkpkg -output-obj -package uri -package tcpip.udp -package tcpip.tcp -package tcpip.stack-direct -package tcpip.ipv4 -package tcpip.icmpv4 -package tcpip.ethif -package tcpip.arpv4 -package tcpip -package nocrypto.mirage -package nocrypto -package mirage-types-lwt -package mirage-types -package mirage-solo5 -package mirage-runtime -package mirage-random -package mirage-net-solo5 -package mirage-logs -package mirage-conduit -package mirage-clock-freestanding -package mirage-bootvar-solo5 -package magic-mime -package lwt -package io-page -package functoria-runtime -package cohttp-mirage -predicates mirage_solo5 key_gen.cmx dispatch.cmx static1.cmx static2.cmx main.cmx -o main.native.o
File "_none_", line 1:
Error: Files /home/mato/.opam/ukvm-4.06.0/lib/mirage-entropy/mirage-entropy.cmxa
       and /home/mato/.opam/ukvm-4.06.0/lib/mirage-os-shim/solo5/mirage-os-shim.cmxa
       make inconsistent assumptions over interface Mirage_OS
Command exited with code 2.
````

I believe this is due to my adding an `OS.Solo5` module in https://github.com/mirage/mirage-solo5/pull/27/commits/61925cfdb20629a4a866123a74d003f73c5921d7, however mirage-os-shim has "here be dragons" written all over it and I don't know what the fix is.

/cc @hannesm Solo5/solo5#245 mirage/mirage-solo5#27

**Comment by hannesm** [comment]
this is due to 4.06.0 doing -keep-locs by default. fixed in #4, for testing an `opam pin add mirage-os-shim https://github.com/hannesm/mirage-os-shim.git#4.06` should solve your issue.

@pqwy i'd be happy to move this repository to the mirageos organisation and maintain it in the future!?

**Comment by mato** [comment]
@hannesm That fixes it, thanks. I'll add a pin for this version to Solo5/opam-solo5 for now.

**Comment by hannesm** [comment]
fixed by 3.0.1 release
